trk/src/Screens/GymDaily/index.js

```diff
@@ -1,8 +1,219 @@
+    type: '',
+    climb: '',
+  });
+  const [latestFeedback, setLatestFeedback] = useState({
+    explanation: '',
+    climb: '',
+    rating: '',
+    climbName: '',
+  });
+
+  const { currentUser } = useContext(AuthContext);
+  const { getClimbsBySomeField } = ClimbsApi();
+  const { getTapsBySomeField } = TapsApi();
+  const { getCommentsBySomeField } = CommentsApi();
+  const { getClimb } = ClimbsApi();
+
+  const userId = currentUser.uid;
+
+
+  useEffect(() => {
+    const fetchClimbs = async () => {
+      try {
+        const querySnapshot = await getClimbsBySomeField('setter', userId);
+        if (querySnapshot && querySnapshot.docs) {
+          const climbs = querySnapshot.docs.map(doc => {
+            return {
+              id: doc.id,
+              ...doc.data()
+            };
+          }).filter(climb => !climb.archived); // Filter out archived climbs
+
+          setYourClimbs(climbs);
+          setTotalClimbs(climbs.length.toString()); // Now reflects the count of non-archived climbs
+        } else {
+          console.error("No data found in Firestore query");
+        }
+      } catch (error) {
+        console.error("Error fetching climbs:", error);
+      }
+    };
+
+    fetchClimbs();
+    console.log(`Your climbs are: ${JSON.stringify(yourClimbs, null, 2)}`);
+
+  }, [userId]);
+
+
+  useEffect(() => {
+    const fetchTapsAndCalculateAscents = async () => {
+      let totalTaps = 0;
+      let allTaps = [];
+      let latestAscentsCount = 0;
+      const oneDayAgo = new Date(Date.now() - 24 * 60 * 60 * 1000);
+      let climbFrequencyMap = {};
+      let timeFrequencyMap = {};
+
+
+
+      for (const climb of yourClimbs) {
+        const tapsSnapshot = await getTapsBySomeField('climb', climb.id);
+        if (tapsSnapshot && tapsSnapshot.docs) {
+          const taps = tapsSnapshot.docs.map(doc => {
+            const tap = doc.data();
+            tap.timestamp = tap.timestamp.toDate();
+            console.log(`Tap Timestamp: ${tap.timestamp}`);
+            return tap;
+          });
+          totalTaps += taps.length;
+          allTaps = [...allTaps, ...taps];
+
+          latestAscentsCount += taps.filter(tap => tap.timestamp > oneDayAgo).length;
+
+          taps.forEach(tap => {
+            if (!climbFrequencyMap[tap.climb]) {
+              climbFrequencyMap[tap.climb] = { count: 0, name: climb.name, id: climb.id, grade: climb.grade };
+            }
+            climbFrequencyMap[tap.climb].count++;
+          });
+        }
+      }
+
+      for (const tap of allTaps) {
+        if (tap.timestamp) {
+          const hour = tap.timestamp.getHours(); 
+          timeFrequencyMap[hour] = (timeFrequencyMap[hour] || 0) + 1;
+        }
+      }
+
+
+      let peakHour = Object.keys(timeFrequencyMap).reduce((a, b) => timeFrequencyMap[a] > timeFrequencyMap[b] ? a : b, null);
+      setPeakTime(peakHour ? `${peakHour}:00 - ${parseInt(peakHour) + 1}:00` : '');
+      console.log(`peakHour is: ${peakHour}`)
+
```

Comment:
here we use the timestamp of all taps and convert it to a readable 1 hour window. we then calculate the most frequent 1hr window which is assigned to the peakHour const